src/Infrastructure.EntityFramework/Auth/Repositories/AuthRequestRepository.cs

```diff
@@ -15,15 +15,19 @@ public class AuthRequestRepository : Repository<Core.Auth.Entities.AuthRequest,
     public AuthRequestRepository(IServiceScopeFactory serviceScopeFactory, IMapper mapper)
         : base(serviceScopeFactory, mapper, (DatabaseContext context) => context.AuthRequests)
     { }
-    public async Task<int> DeleteExpiredAsync()
+    public async Task<int> DeleteExpiredAsync(TimeSpan userExpiration, TimeSpan adminExpiration, TimeSpan adminApprovalExpiration)
     {
+
         using (var scope = ServiceScopeFactory.CreateScope())
         {
             var dbContext = GetDatabaseContext(scope);
-            var expiredRequests = await dbContext.AuthRequests.Where(a => a.CreationDate < DateTime.Now.AddMinutes(-15)).ToListAsync();
+            var expiredRequests = await dbContext.AuthRequests
+                .Where(a => (a.Type != AuthRequestType.AdminApproval && a.CreationDate.AddSeconds(userExpiration.TotalSeconds) < DateTime.UtcNow)
+                    || (a.Type == AuthRequestType.AdminApproval && a.Approved != false && a.CreationDate.AddSeconds(adminExpiration.TotalSeconds) < DateTime.UtcNow)
```

Comment:
❓ : SQL checks for `([Approved] IS NULL OR [Approved] = 0)` but this query just checks for false. Should it also remove nulls to match the SQL behavior?

Reply:
EF is smart enough to recognize that it is nullable and generate the proper SQL for if that language needs a special check for null or not. For example this is the SQL that is generated for Sqlite:
```sql
SELECT "a"."Id", "a"."AccessCode", "a"."Approved", "a"."AuthenticationDate", "a"."CreationDate", "a"."Key", "a"."MasterPasswordHash", "a"."OrganizationId", "a"."PublicKey", "a"."RequestDeviceIdentifier", "a"."RequestDeviceType", "a"."RequestIpAddress", "a"."ResponseDate", "a"."ResponseDeviceId", "a"."Type", "a"."UserId"
FROM "AuthRequest" AS "a"
WHERE ((("a"."Type" <> 2) AND (rtrim(rtrim(strftime('%Y-%m-%d %H:%M:%f', "a"."CreationDate", CAST(@__userExpiration_TotalSeconds_0 AS TEXT) || ' seconds'), '0'), '.') < rtrim(rtrim(strftime('%Y-%m-%d %H:%M:%f', 'now'), '0'), '.'))) OR ((("a"."Type" = 2) AND (("a"."Approved" <> 0) OR ("a"."Approved" IS NULL))) AND (rtrim(rtrim(strftime('%Y-%m-%d %H:%M:%f', "a"."CreationDate", CAST(@__adminExpiration_TotalSeconds_1 AS TEXT) || ' seconds'), '0'), '.') < rtrim(rtrim(strftime('%Y-%m-%d %H:%M:%f', 'now'), '0'), '.')))) OR ((("a"."Type" = 2) AND ("a"."Approved" = 1)) AND (rtrim(rtrim(strftime('%Y-%m-%d %H:%M:%f', "a"."ResponseDate", CAST(@__adminApprovalExpiration_TotalSeconds_2 AS TEXT) || ' seconds'), '0'), '.') < rtrim(rtrim(strftime('%Y-%m-%d %H:%M:%f', 'now'), '0'), '.')))
```

It's kinda insane to read but buried in there is `(("a"."Approved" <> 0) OR ("a"."Approved" IS NULL))` it decided to alternate it but should work all the same. But I don't fully trust every provider to do it properly hence the tests. But this is a good shout, I should also have a test the has an AdminApproval that was denied and make sure that it gets deleted after the 7 days.

Reply:
Ah, thank you for the explanation. Glad it helped id another necessary test case!

Reply:
This helped me find a bug in the EF implementation, as you can see above what Sqlite was generating for that check it was NOT logically equivalent to the SQL implementation. I fixed the bug and now it will generate `(("a"."Approved" <> 1) OR ("a"."Approved" IS NULL))`, which is different but now logically equivalent. 

Do you think I should change SQL to be ([Approved] IS NULL OR [Approved] != 1) though?